app/modules/tcs34725.c

```diff
@@ -0,0 +1,359 @@
+	TCS34725_GAIN_4X								= 0x01,	 /**<	2x gain	*/
+	TCS34725_GAIN_16X							 = 0x02,	 /**<	16x gain */
+	TCS34725_GAIN_60X							 = 0x03		/**<	60x gain */
+}
+tcs34725Gain_t;
+static void temp_setup_debug(int line, const char *str);
+uint8_t tcs34725Setup(lua_State* L);
+uint8_t tcs34725Enable(lua_State* L);
+uint8_t tcs34725Disable(lua_State* L);
+uint8_t tcs34725GetRawData(lua_State* L);
+uint8_t tcs34725LuaSetIntegrationTime(lua_State* L);
+uint8_t tcs34725SetIntegrationTime(tcs34725IntegrationTime_t it, lua_State* L);
+uint8_t tcs34725LuaSetGain(lua_State* L);
+uint8_t tcs34725SetGain(tcs34725Gain_t gain, lua_State* L);
+
+static bool							_tcs34725Initialised = false;
+static int32_t						_tcs34725SensorID = 0;
+static tcs34725Gain_t				_tcs34725Gain = TCS34725_GAIN_1X;
+static tcs34725IntegrationTime_t	_tcs34725IntegrationTime = TCS34725_INTEGRATIONTIME_2_4MS;
+
+os_timer_t tcs34725_timer; // timer for forced mode readout
+sint32_t cb_tcs_en;
+sint32_t self_ref;
+
+/**************************************************************************/
+/*!
+		@brief	Writes an 8 bit values over I2C
+*/
+/**************************************************************************/
+uint8_t tcs34725Write8 (uint8_t reg, uint8_t value)
+{
+	platform_i2c_send_start(TCS34725_BUS_ID);
+	platform_i2c_send_address(TCS34725_BUS_ID, TCS34725_ADDRESS, PLATFORM_I2C_DIRECTION_TRANSMITTER);
+	platform_i2c_send_byte(TCS34725_BUS_ID, TCS34725_COMMAND_BIT | reg );
+	platform_i2c_send_byte(TCS34725_BUS_ID, value);
+	platform_i2c_send_stop(TCS34725_BUS_ID);
+	return 0;
+}
+
+/**************************************************************************/
+/*!
+		@brief	Reads a 8 bit values over I2C
+*/
+/**************************************************************************/
+uint8_t tcs34725Read8(uint8_t reg)
+{
+	uint8_t value;
+	
+	platform_i2c_send_start(TCS34725_BUS_ID);
+	platform_i2c_send_address(TCS34725_BUS_ID, TCS34725_ADDRESS, PLATFORM_I2C_DIRECTION_TRANSMITTER);
+	platform_i2c_send_byte(TCS34725_BUS_ID, TCS34725_COMMAND_BIT | reg);
+	platform_i2c_send_stop(TCS34725_BUS_ID);
+	
+	platform_i2c_send_start(TCS34725_BUS_ID);
+	platform_i2c_send_address(TCS34725_BUS_ID, TCS34725_ADDRESS, PLATFORM_I2C_DIRECTION_RECEIVER);
+	value = platform_i2c_recv_byte(TCS34725_BUS_ID, 0);
+	platform_i2c_send_stop(TCS34725_BUS_ID);
+	
+	return value;
+}
+
+/**************************************************************************/
+/*!
+		@brief	Reads a 16 bit values over I2C
+*/
+/**************************************************************************/
+uint16_t tcs34725Read16(uint8_t reg)
+{		
+	uint8_t low = tcs34725Read8(reg);
+	uint8_t high = tcs34725Read8(++reg);
+	
+	return (high << 8) | low;
+}
+/**************************************************************************/
+/*!
+		@brief	Finishes enabling the device
+*/
+/**************************************************************************/
+uint8_t tcs34725EnableDone(lua_State* L)
```

Comment:
Don't use cached Lua state in a callback function.
Best practice is to get a Lua state with
```c
lua_State *L = lua_getstate()
```

Refer to https://github.com/nodemcu/nodemcu-firmware/issues/846#issuecomment-164651561 for details.